code/headers/ssd1306_oled_buffered.hpp

```diff
@@ -10,52 +10,90 @@ namespace r2d2::display {
      * SSD1306 buffered interface for an oled
      * Implements hwlib::window to easily use text and drawing functions that
      * are already implemented. Extends from r2d2::display::ssd1306_i2c_c
+     * 
+     * The template parameters are used for the parent class.
      */
-    class ssd1306_oled_buffered_c : public ssd1306_i2c_c {
+    template<std::size_t Cursor_Count, uint8_t Display_Size_Width, uint8_t Display_Size_Height>
+    class ssd1306_oled_buffered_c : public ssd1306_i2c_c<Cursor_Count, Display_Size_Width, Display_Size_Height> {
     private:
         /**
          * The buffer with the pixel data
          * The first byte is used for the data-prefix that the display driver
          * requires to be sent when sending pixel data.
          */
-        uint8_t buffer[(width * height / 8) + 1] = {};
+        uint8_t buffer[(Display_Size_Width * Display_Size_Height / 8) + 1] = {};
```

Comment:
Why is the buffer size dependent on the input templates? The display has a fixed width and height. It is not magically gonna change because you make the buffer smaller.

Reply:
This is not true, because a single chip can have different display dimensions. The display currently requires a big change to work with offsets and screen widths, but that is out of the scope of this pr, and would not be correctly placed in this branch.